src/common/snippets/include/snippets/lowered/expression.hpp

```diff
@@ -19,12 +19,14 @@ namespace lowered {
 
 class LinearIR;
 using ExpressionPtr = std::shared_ptr<Expression>;
+using ExressionMap = std::unordered_map<Expression*, ExpressionPtr>;
 class Expression : public std::enable_shared_from_this<Expression> {
     friend class LinearIR;
     friend class ExpressionPort;
 
 public:
     Expression() = default;
+    Expression(const Expression& other);
```

Comment:
We can, but note that it will make `Expressions` non-copyable outside of `Expression::` or friendly classes. This is not a bad thing, since copying generally produces invalid expression (because internal pointers are not updated), but we should keep it in mind.